### ISSUE TITLE
fix: dart frog can't read env variables passed on command line.

### ISSUE THREAD

**Issue by erickzanardo** [VeryGoodOpenSource/dart_frog#557]
**Description**

As a developer I want to be able to pass environment variables when running the server in dev mode, or the built one, passing those variables in the command. For example:

```
FOO=BAR dart_frog dev
```
or
```
FOO=BAR dart build/bin/server.dart
```

```dart
import 'package:dart_frog/dart_frog.dart';

Response onRequest(RequestContext context) {
  return Response(
    body: 'Welcome to Dart Frog!${String.fromEnvironment('FOO')}',
  );
}
```

**Steps To Reproduce**

1. Create a server with an endpoint with the above snippet
2. Try running the server with the commands above
3. The env var is empty.

**Expected Behavior**

The env var should have been "rendered" on the response.

**Comment by felangel** [comment]
Currently, when env variables are specified you can access them via `Platform.environment`

**Comment by erickzanardo** [comment]
Like @felangel mentioned, using `Platform.environment` works. I got curious on why `Platform.environment` while `String.fromEnvironment` doesn't, so I went on to understand why.

It seems that `String.fromEnvironment` only reads declarations provided by the compilation or interpretation process (like the dart-define used on Flutter)

So this is a non-issue, just a confusion from my part.

With that said, I believe this is a confusion that may occur to more people, so I wonder if we should have something in our docs mentioning that `String.fromEnvironment` shouldn't be used in a Dart frog app. Wdyt?
